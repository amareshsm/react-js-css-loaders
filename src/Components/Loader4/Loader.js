import React from 'react'
import styles from './style.css'
const Loader4 = (props) => (
  <>
    <style>{`${styles}`}</style>
    <div className="loaders">
      <div className="loader-div four">
        <div className="loader4"></div>
      </div>
    </div>
  </>
)

export default Loader4
