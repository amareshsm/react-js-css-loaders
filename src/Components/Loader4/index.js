export { default as Loader4 } from './Loader'
