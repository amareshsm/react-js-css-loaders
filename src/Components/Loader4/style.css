.loaders {
    margin: 20px;
    display: flex;
    flex-wrap: wrap;
    box-sizing: border-box;
    justify-content: space-between;
}

.loader-div.four {
    background-color: pink;
}

.loader4,
.loader4::before,
.loader4::after {
    background-color: black;
    width: 10px;
    height: 30px;
    position: relative;
    animation: loader4 1.8s ease-in-out infinite;
}

.loader4 {
    animation-delay: -0.16s;
    transform: translateZ(0);
    color: black;
}

.loader4::before,
.loader4::after {
    position: absolute;
    content: "";
    top: 0;
}

.loader4::before {
    left: -14px;
    animation-delay: -0.32s;
}
.loader4::after {
    left: 14px;
}

@keyframes loader4 {
    0%,
    80%,
    100% {
        height: 35px;
        box-shadow: 0 0;
    }

    40% {
        height: 55px;
        box-shadow: 0 -10px;
    }
}
