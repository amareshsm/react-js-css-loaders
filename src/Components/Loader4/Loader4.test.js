import React from 'react'
import renderer from 'react-test-renderer'
import { Loader4 } from '../Loader4'
it('renders correctly', () => {
  const tree = renderer.create(<Loader4 />).toJSON()
  expect(tree).toMatchSnapshot()
})
