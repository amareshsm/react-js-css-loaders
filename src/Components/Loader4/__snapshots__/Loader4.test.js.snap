// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <style>
    [object Object]
  </style>,
  <div
    className="loaders"
  >
    <div
      className="loader-div four"
    >
      <div
        className="loader4"
      />
    </div>
  </div>,
]
`;
